openmc/data/kalbach_mann.py

```diff
@@ -0,0 +1,402 @@
+from collections.abc import Iterable
+from numbers import Real, Integral
+from warnings import warn
+
+import numpy as np
+
+import openmc.checkvalue as cv
+from openmc.stats import Tabular, Univariate, Discrete, Mixture
+from .function import Tabulated1D, INTERPOLATION_SCHEME
+from .angle_energy import AngleEnergy
+from .data import EV_PER_MEV
+from .endf import get_list_record, get_tab2_record
+
+
+class KalbachMann(AngleEnergy):
+    """Kalbach-Mann distribution
+
+    Parameters
+    ----------
+    breakpoints : Iterable of int
+        Breakpoints defining interpolation regions
+    interpolation : Iterable of int
+        Interpolation codes
+    energy : Iterable of float
+        Incoming energies at which distributions exist
+    energy_out : Iterable of openmc.stats.Univariate
+        Distribution of outgoing energies corresponding to each incoming energy
+    precompound : Iterable of openmc.data.Tabulated1D
+        Precompound factor 'r' as a function of outgoing energy for each
+        incoming energy
+    slope : Iterable of openmc.data.Tabulated1D
+        Kalbach-Chadwick angular distribution slope value 'a' as a function of
+        outgoing energy for each incoming energy
+
+    Attributes
+    ----------
+    breakpoints : Iterable of int
+        Breakpoints defining interpolation regions
+    interpolation : Iterable of int
+        Interpolation codes
+    energy : Iterable of float
+        Incoming energies at which distributions exist
+    energy_out : Iterable of openmc.stats.Univariate
+        Distribution of outgoing energies corresponding to each incoming energy
+    precompound : Iterable of openmc.data.Tabulated1D
+        Precompound factor 'r' as a function of outgoing energy for each
+        incoming energy
+    slope : Iterable of openmc.data.Tabulated1D
+        Kalbach-Chadwick angular distribution slope value 'a' as a function of
+        outgoing energy for each incoming energy
+
+    """
+
+    def __init__(self, breakpoints, interpolation, energy, energy_out,
+                 precompound, slope):
+        super().__init__()
+        self.breakpoints = breakpoints
+        self.interpolation = interpolation
+        self.energy = energy
+        self.energy_out = energy_out
+        self.precompound = precompound
+        self.slope = slope
+
+    @property
+    def breakpoints(self):
+        return self._breakpoints
+
+    @property
+    def interpolation(self):
+        return self._interpolation
+
+    @property
+    def energy(self):
+        return self._energy
+
+    @property
+    def energy_out(self):
+        return self._energy_out
+
+    @property
+    def precompound(self):
+        return self._precompound
+
+    @property
+    def slope(self):
+        return self._slope
+
+    @breakpoints.setter
+    def breakpoints(self, breakpoints):
+        cv.check_type('Kalbach-Mann breakpoints', breakpoints,
```

Comment:
Moving `'Kalbach-Mann'` to a variable will allow its reuse and prevent any bugs from typos. Most IDEs will autocomplete a variable.


```
KALBACH_MANN = 'Kalbach-Mann'
cv.check_type(KALBACH_MANN + ' breakpoints', breakpoints, .... )
```


This can be reused in other functions as well